### PR TITLE
WIP: Enable custom html in iframe <head>

### PR DESCRIPTION
**Pull request by karl**
This is a work in progress to enable adding custom html to the iframe that the components are rendered in.

This is useful for loading fonts from Typekit (which require specific script tags).

The idea was brought up in this issue: https://github.com/kadirahq/react-storybook/issues/61

### CODE REVIEW

**Comment by karl** [comment]
There are currently no tests or documentation for this. I just wanted a sanity check that I'm on the right path before spending too long on this.

Any and all feedback on this is much appreciated!

**Comment by jeef3** [comment]
Cool, looks like a reasonable way of doing it. Just allow the user to supply their own `head.html` and that gets injected.

There's probably a nicer way of doing it, but I think this would go a long way to helping :+1:

File: src/server/iframe.html.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
-export default function () {
+export default function (extraHtml) {
```
Comment (arunoda) [inline src/server/iframe.html.js]: May be rename as `extraHead`

**Comment by arunoda** [comment]
Yes.This is the way it should be. We don't have any tests for the server.
It's great, if you can start it with this PR. (Just for this case).

**Comment by karl** [comment]
@arunoda I've added tests for the new functionality. I'm inexperienced with testing JS on the server, and wasn't sure how best to test the function. I included `mock-fs` to mock the file system while testing, let me know if that is not the preferred approach!

I'll look at adding some documentation next.

File: src/server/get_head_html.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import path from 'path';
```
Comment (arunoda) [inline src/server/get_head_html.js]: Shall we move this into a file called `src/server/utils.js` and export the following function as `getHeadHtml`.

**Comment by arunoda** [comment]
@karl Everything looks great.
Once you move the content as I suggested, we are good to go.

**Comment by arunoda** [comment]
I've done the suggested I made and merged.

**Comment by karl** [comment]
Awesome, thanks @arunoda!

**Comment by arunoda** [comment]
Thanks you too.

**Pull request merged**